packages/ui/src/components/ProjectCard/ProjectCard.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Box, Flex, forwardRef, Grid } from '@chakra-ui/react';
+
+import { Button } from '../Button/Button';
+import { Typography } from '../Typography/Typography';
+
+export type StatusProps = 'idle' | 'doing' | 'review' | 'done';
+
+type RemoveCommonValues<Type, TOmit> = {
+  [Property in keyof Type]: TOmit extends Record<Property, infer U> ? Exclude<Type[Property], U> : Type[Property];
+};
+
+type Id<Type> = Record<string, unknown> & { [P in keyof Type]: Type[P] };
+
+type ConditionalProps<Type, TKey extends keyof TCase, TCase extends Partial<Type>> =
+  | Id<Omit<Type, keyof TCase> & TCase>
+  | Id<RemoveCommonValues<Type, Pick<TCase, TKey>>>;
+
+export type ProjectCardType = ConditionalProps<
+  {
+    status: StatusProps;
+    image: string;
+    title: string;
+    url: string;
+  },
+  'status',
+  | {
+      status: 'idle' | 'doing' | 'review';
+      date: Date;
+    }
+  | {
+      status: 'done';
+      points: number;
+      pointsMax: number;
+    }
+>;
+
+export const ProjectCard = forwardRef<ProjectCardType, 'div'>(
+  // disabled for url props
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  ({ status, image, title, url, points, pointsMax, date, ...props }, ref) => {
+    const dateProps = date as Date;
+    let statusText;
+    let timeStatus;
+    let dateOrPointsText;
+    let isDoing = true;
+
+    if (status === 'idle') {
+      statusText = 'Nie rozpoczęty';
+      timeStatus = 'Rozpoczecie';
+      dateOrPointsText = dateProps.toLocaleDateString();
+      isDoing = false;
+    }
+    if (status === 'doing') {
+      statusText = 'W trakcie';
+      timeStatus = 'Deadline';
+      dateOrPointsText = dateProps.toLocaleDateString();
+    }
+    if (status === 'review') {
+      statusText = 'W ocenie';
+      timeStatus = 'Demo';
+      dateOrPointsText = dateProps.toLocaleDateString();
+    }
+    if (status === 'done') {
+      statusText = 'Zakończony';
+      timeStatus = 'Punkty';
+      dateOrPointsText = `${points}/${pointsMax}`;
+    }
+
```

Comment:
🏕️ So in case of performance we should replace these if statements with IMO switch case, or if you don't want to, use else if, this way performance is better, right now it's going through and checking every if statement even if previous check passed
I'm also a fan of extracting such a helpers outside of the component, but it's up to you :D
And also, for future purposes I think we should make an enum for "status" props and export it, I don't think that using in future plain strings as a props is good and maintainable idea

Reply:
I agree it would be better to move this to a separate function outside of the component. In such a case you can as well use if's with early returns.

When it comes to the enums idea - we have enums usage listed as a bad practice in our docs. Of course @Veryyapeee  if you want to discuss this topic we can do it.

Reply:
As Konrad noted, we do not use enums in this project. I will leave status type as it is.